source/rds.rst

```diff
@@ -0,0 +1,202 @@
+
+See below for some detailed documentation on using the Postgres command
+line to do the things that will need to be done.
+
+Backups
+-------
+
+TBD!
+
+(Do we just rely on the automatic backups on RDS, which are only kept up
+to 45 days; do we make "manual" (initiated when we choose) backups on RDS,
+which will be kept as long as we want; or do we want to make backups
+outside of RDS?)
+
+Practical tips
+~~~~~~~~~~~~~~
+
+Before starting to work with RDS, read or at least skim through
+`the AWS documentation on how Postgres works on RDS
+<http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_PostgreSQL.html>`_,
+and pay special attention to
+`Appendix: Common DBA Tasks for PostgreSQL <http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Appendix.PostgreSQL.CommonDBATasks.html>`_.
+
+It'll save you a lot of headaches in the long run.
+
+Postgres on RDS
+---------------
+
+The master user that RDS gives you is *not* a Postgres superuser. Though it
+has a lot of permissions, there are a few things that you just won't be able
+to do quite the way you could if you were a superuser.
+
+In the examples below, for readability I'm omitting most of the common
+arguments to specify where the Postgres server is, what the database name is,
+etc. You can set some environment variables to use as defaults for things::
+
+    export PGDATABASE=dbname
+    export PGHOST=xxxxxxxxx
+    export PGUSER=master
+    export PGPASSWORD=xxxxxxxxxx
+
+PGPASSWORD behaves the same as password connection parameter. Use of this
+environment variable is not recommended for security reasons (some operating
+systems allow non-root users to see process environment variables via ps);
+instead consider using the ~/.pgpass file (see Section 30.14 of the PG docs).
+
+Create user
+-----------
+
+This is pretty standard::
+
+    export PGUSER=master
+    createuser <username>
+    psql -c "ALTER USER <username> WITH PASSWORD 'plaintextpassword';"
+
+Create database
+---------------
+
+If you need to create a database owned by ``project_user``, you'll need
+to create it as ``master`` and then modify the ownership and permissions::
+
+    export PGUSER=master
+    createdb --template=template0 'dbname'
+    psql -c "revoke all on database 'dbname' from public;"
+    psql -c "grant all on database 'dbname' to project_user;"
+
+If you need to enable extensions etc, do that now (see below).  When done, then::
+
+    psql -c "alter database 'dbname' owner to project_user;"
+
+A superuser could create the database already owned by a specific user,
+but RDS's master user cannot.
+
+PostGIS
+-------
+
+To enable PostGIS, as the master user::
+
+    export PGUSER=master
+    psql -c "create extension postgis;"
+    psql -c "alter table spatial_ref_sys OWNER TO $PROJECT_USER;"
+
+where ``$PROJECT_USER`` is the postgres user who will be using the database.
+
+(Outside of RDS, only a superuser can use ``create extension``; RDS has special
+handling for a whiteless of extensions.)
```

Comment:
Is that supposed to be 'whitelist'? or is it just a term I haven't learned yet...